src/widgets/airplanepainter.cpp

```diff
@@ -0,0 +1,56 @@
+#include "airplanepainter.hpp"
+#include <QPainter>
+#include <cmath>
+
+// paint origin
+const QPoint origin(0,0);
+// size of plane
+const QSize planeSize(26,26);
+// Used to center painter while rotating painter mid-point of plane widget
+const QPoint translateCenter(planeSize.width()/2,planeSize.height()/2);
+// set's value back to original point if translateCenter was used to translate painter for painting plane widget
+const QPoint translateReset(planeSize.width()/2-planeSize.width(),planeSize.height()/2-planeSize.height());
+
+/* Constructor */
+AirplanePainter::AirplanePainter(QWidget *parent)
+   : QWidget(parent), m_plane(":/res/airplane.png")
+{}
+
+/**
+ * Draws plane at the angle within angleValue
+ */
+void AirplanePainter::paintEvent(QPaintEvent*)
+{
+    QPainter painter;
+    painter.begin(this);
+
+    /* Rotate plane at the center of the widget */
+    painter.translate(translateCenter);
+    painter.rotate(angleValue);
+    painter.translate(translateReset);
+
+    // Paint m_plane at `origin` (x:0,y:0), size `planeSize` (26 x 26)
+    painter.drawPixmap(QRect(origin,planeSize), m_plane);
+
+    painter.end();
```

Comment:
This line is probably unnecessary as well

Edit: yes it is redundant, so remove it
https://doc.qt.io/qt-5/qpainter.html#end